lib/beaker/host_prebuilt_steps.rb

```diff
@@ -289,7 +292,14 @@ def get_ip(host)
     # @param [Host] host the host to act upon
     # @param [String] etc_hosts The string to append to the /etc/hosts file
     def set_etc_hosts(host, etc_hosts)
-      host.exec(Command.new("echo '#{etc_hosts}' > /etc/hosts"))
+      if host['platform'] =~ /freebsd/
+        # FreeBSD gets weird about special characters, we have to go a little OTT here
+        freebsd_hosts = etc_hosts.gsub!(/\t/,'\\t')
+        freebsd_hosts = etc_hosts.gsub!(/\n/,'\\n')
```

Comment:
`gsub!` modifies the `etc_hosts` parameter that the caller passes into this method, and would be very surprising.